src/modules/accommodations/accommodations.controller.ts

```diff
@@ -23,4 +43,36 @@ export class AccommodationsController {
       limit: searchQueryParams.limit,
     };
   }
+
+  @Post()
+  @CreateAccommodationSwaggerDecorator()
+  async createAccommodation(@Body() createDto: CreateAccommodationDto) {
+    return this.accommodationsService.createAccommodation(createDto);
+  }
+
+  @Get(':id')
+  @GetByIdSwaggerDecorator()
+  async getAccommodationById(@Param('id') id: string) {
+    return this.accommodationsService.getAccommodationById(id);
+  }
+
+  @Put(':id')
+  @UpdateAccommodationSwaggerDecorator()
+  async updateAccommodation(
+    @Param('id') id: string,
+    @Body() updateDto: UpdateAccommodationDto,
+  ) {
+    return this.accommodationsService.updateAccommodation(id, updateDto);
+  }
+
+  @Delete(':id')
+  @DeleteAccommodationSwaggerDecorator()
+  async deleteAccommodation(@Param('id') id: string) {
+    const accommodation =
+      await this.accommodationsService.getAccommodationById(id);
+    if (!accommodation) {
+      throw new NotFoundException(`Accommodation with ID ${id} not found`);
+    }
+    return this.accommodationsService.deleteAccommodation(id);
```

Comment:
deleteAccommodation does not return anything. you are just returning empty response right now. I recommend not to return anything and just return status code 204 on successful deletion. Use  @HttpCode(204) from nest.